receiver/hostmetricsreceiver/scraper/scraper.go

```diff
@@ -0,0 +1,67 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scraper
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+)
+
+// Scraper gathers metrics from the host machine and converts
+// these into internal metrics format.
+type Scraper interface {
```

Comment:
Do we need this to be publicly exported? Consider adding it in a "internal" directory for the moment. Smaller the public interface easier to maintain.

Also what is the rational to not have this as un-exported type directly in the hostmetricsreceiver? We can start with that for the moment and see if we really need to ever export this.

Reply:
In the scraper packages, when they exist, they will need to reference this `Scraper` interface, e.g. in the `Factory.CreateMetricsScraper` function.

In the hostmetricsreceiver package, `hostmetrics_receiver_test` will need to reference scraper packages, e.g.
```
config := &Config{
  Scrapers: map[string]scraper.Config{
    cpuscraper.TypeStr: &cpuscraper.Config{ ... },
  },
}
```

So I had to pull these interfaces into a separate package to avoid a cicular dependency.

I can move these definitions into an "internal" package, but I'll need to at least leave the `MakeScraperFactoryMap` function available externally, which is used to inject the factories in `defaults.go`. Alternatively, I could just call this function from inside `hostmetricsreceiver.Factory` directly; I'd rather let it be injected, but I'm not sure if that goes against the design of receivers being completely self contained?

Reply:
Question is, do we need to support a public mechanism to inject external implementations, or we say that we support just a set of scrapers that we implement and maintain.

The reason is that this is very confusing between scraper and receiver. When do I implement one or the other.